### PR TITLE
Refactored userdata

### PR DESCRIPTION
**Pull request by timidri**
- moved important user data templates to templates/ directory and removed from common.yaml
- create_host.pp now supports 2 options for specifying user data:
  - either pass `user_data_template` parameter which contains the template file name, like `awskit/linux_userdata.epp`. Then, the template is processed by create_host using default parameters.
  - or pass the `user_data` parameter containing the user data itself. In this case, create_host will not attempt to process the user data and will use it as is
- moved the boltws-specific user data to inline templates inside the class `awskit::create_bolt_workshop_targets`
- some code cleanup

### CODE REVIEW

**Comment by maju6406** [comment]
@timidri sorry about the delay on this. Looks like there are some conflicts now. Want to resolve them, and then I'll take a look?

**Comment by timidri** [comment]
> @timidri sorry about the delay on this. Looks like there are some conflicts now. Want to resolve them, and then I'll take a look?

I've rebased using the new master, should have no conflicts now.

**Comment by maju6406** [comment]
@timidri one last conflict and I should be ready to merge

**Comment by timidri** [comment]
@maju6406 Rebased again, now hopefully also fixed the travis-ci issues 👍 
@eboutili Please be aware that this PR drastically changes the userdata system, and it unfortunately will break your userdata customization for create-dockerhost.pp. Let's get together to discuss a new approach to customization. See also CHANGELOG.md for the description of the changes.

**Comment by timidri** [comment]
@maju6406 Hi, any chance you can review this PR? Thanks!

**Comment by maju6406** [comment]
@timidri Looks like there's 2 conflicting files still.

File: manifests/create_bolt_workshop_master.pp [hunk 1]
```diff
@@ -36,7 +36,6 @@
   awskit::create_host { $instance_name:
     ami             => $awskit::centos_ami,
     instance_type   => $instance_type,
-    user_data       => '',
```
Comment (kreeuwijk) [inline manifests/create_bolt_workshop_master.pp]: I had to specify something for user_data in the original setup; is user_data_template now truly optional?

File: manifests/create_bolt_workshop_targets.pp [hunk 2]
```diff
@@ -37,40 +35,48 @@
     ],
   }
 
-  #Create the Linux target for teacher
```
Comment (kreeuwijk) [inline manifests/create_bolt_workshop_targets.pp]: You are removing the deployment of the Bolt targets for the trainer, that is not acceptable

Comment (timidri) [reply manifests/create_bolt_workshop_targets.pp]: @kreeuwijk hmm, strange, not sure what happened there. Resurrecting...

File: manifests/create_bolt_workshop_master.pp (hunk 1 above)
Comment (timidri) [reply manifests/create_bolt_workshop_master.pp]: @kreeuwijk There are the define parameters now:
```
define awskit::create_host (
  $ami,
  $instance_type,
  $user_data          = undef,
  $user_data_template = undef,
  $master_ip          = undef,
  $master_name        = undef,
  $run_agent          = true,
  $security_groups    = 'none',
  $key_name           = 'none',
  $role               = undef,
  $environment        = undef,
  $public_ip          = undef,
){
```
Both `$user_data` and `$user_data_template` are optional now. To be more sure of the correct workings, you should add spec tests for your new bolt-workshop-related classes.

**Pull request merged**